lib/chained_job/process.rb

```diff
@@ -22,13 +22,22 @@ def run
       with_hooks do
         return finished_worker unless argument
 
-        job_instance.process(argument)
+        begin
+          job_instance.process(argument)
+        rescue StandardError => e
+          push_job_arguments_back if handle_retry?
```

Comment:
chain would be terminated, even when handle_retry? is true, You probably wanted:

```
if handle_retry?
  push_job_arguments_back : 
else 
  raise e
end
```

Reply:
No no - this would be bad. If we handle exceptions and not raising back, the worker won't' get to retry the queue and it's going completely to stop the processing chain. Let's say you have parallelism => 4. So if all 4 workers receive errors and we are not raising back - the chain is stopped. Right now, before the change it goes back to retry and just picks next args from the Redis and continues with the chain.

Reply:
I mean, relying on sidekiq retries is not a god idea. We should rely on proceeding with a chain. Some jobs have retry set to zero, so if you raise an error in chain the line https://github.com/vinted/chained_job/blob/c980bdbe20f3d3c27cbfc9796670c9531e710009/lib/chained_job/process.rb#L31 would not be reached.

Reply:
That's ok. It should work like this.

If the job is defined with retry:0 - you would still be able to get an exception down in the #process method and again this line would be newer reached. The worker just shutdowns and that's it. If all parallel worker gets the error - all going to shut down and chain is stopped. 

Of course with retry:0 - usually, inside #process developers should handle some errors by logic when. If not - then all workers have the potential to silently shutdown. This change has no effect on this line.

Reply:
In general, this also could work, instead of going into the retry queue, this worker completes but starts a new one. Well it's not ideally what I was planned. 

```
if handle_retry?
  push_job_arguments_back
else 
  raise e
end
```

Reply:
Alright